src/bms/parse.rs

```diff
@@ -110,14 +110,13 @@ impl Bms {
     }
 
     /// Parses a token stream into [`Bms`] without AST.
-    pub fn from_token_stream_without_ast<'a>(
-        token_iter: impl Into<TokenRefIter<'a>>,
+    pub fn from_token_stream_without_ast<'a, T: Iterator<Item = &'a TokenWithPos<'a>>>(
+        token_iter: impl Into<TokenIter<'a, T>>,
```

Comment:
How about using `IntoIterator<Item = &'a TokenWithPos<'a>>>` instead of `Into<TokenIter<'a, T>>`? It's useful for userland to apply other data structures.